examples/solvers/functional_basic_example_solver.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Basic example on how to use the functional class together with solvers.
+
+This file shows an example of how to set up and solve an optimization problem
+using the default functionals. The problem we will solve is to minimize
+1/2 * ||x - g||_2^2 + lam*||x||_1, for some vector g and some constant lam,
+subject to that all components in x are greater than or equal to 0. The
+theoretical optimal solution to this problem is x = (g - lam)_+, where ( )_+
+denotes the positive part of the element, i.e., (z_i)_+ = max(z_i, 0)."""
+
+import numpy as np
+import odl
+
+# Create a space with dimensiona n=10.
+n = 10
+space = odl.rn(n)
+
+# Create parameters.
+g = space.element(np.hstack((np.ones(n/2), -np.ones(n/2))))
```

Comment:
I need to cast `(n/2)` to `int`, otherwise I get
```python
>>> n=10
>>> [0] * (n/2) + [1] * (n/2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't multiply sequence by non-int of type 'float'
```